jena-base/src/test/java/org/apache/jena/atlas/lib/cache/CacheSimpleTest.java

```diff
@@ -40,4 +40,20 @@ public void testFixedSize() {
 				.forEach(testCache::put);
 		assertEquals("Test cache failed to maintain fixed size!", maxSize, testCache.size());
 	}
+
+	@Test
+	public void testSameHash() {
+		Object key1 = new Object() {
```

Comment:
The eventual call to `.equals()` in the main code falls through to `==` (which certainly passes the test), and it might be nice to make this even a little more explicit by overriding `.equals()` in these two guys and making it clear that they are not `equal()` each other, but it's not really necessary.

Reply:
It's indeed a good practice to override `equals` when you change `hashCode` and vice versa.